### PR TITLE
Permit shared-library for libbacktrace

### PR DESCRIPTION
**Pull request by knoepfel**
Just adding a variant and adjusting the corresponding configure args.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libbacktrace/package.py [hunk 1]
```diff
@@ -17,7 +17,14 @@ class Libbacktrace(AutotoolsPackage):
...
 
+    variant("shared", default=False, description="Additionally build shared library")
+
     depends_on("autoconf", type="build")
     depends_on("automake", type="build")
     depends_on("libtool", type="build")
     depends_on("m4", type="build")
+
+    def configure_args(self):
+        if self.spec.variants["shared"]:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libbacktrace/package.py]: The way to check variant values for a spec is to use `self.spec.satisfies("+shared")` in this case.

See the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints, especially note:

"The difference between the two functions is that `satisfies()` tests whether spec constraints overlap at all, while `in` tests whether a spec or any of its dependencies satisfy the provided spec."

File: var/spack/repos/builtin/packages/libbacktrace/package.py [hunk 2]
```diff
@@ -17,7 +17,14 @@ class Libbacktrace(AutotoolsPackage):
...
+            return ["--enable-shared"]
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libbacktrace/package.py]: The Autotools package also has some helper functions you could use.  See the docs at https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#helper-functions.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
At a minimum please use the `satisfies()` method for querying the spec.  You could simplify things using an AutotoolsPackage helper too.

**Comment by knoepfel** [comment]
Thanks, @tldahlgren, for the suggestions.  PR updated accordingly.  Let me know if other changes are requested.

File: var/spack/repos/builtin/packages/libbacktrace/package.py (hunk 2 above)
Comment (knoepfel) [reply var/spack/repos/builtin/packages/libbacktrace/package.py]: Done.

File: var/spack/repos/builtin/packages/libbacktrace/package.py (hunk 1 above)
Comment (knoepfel) [reply var/spack/repos/builtin/packages/libbacktrace/package.py]: Favored Autotools helper.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**